### ISSUE TITLE
Duplicate "component" JSON key in log entries

### ISSUE THREAD

**Issue by kika** [oxidecomputer/omicron#6844]
Maybe it's intentional, but I can't think of a reason. The "component" key in the log entries in a simulated Omicron environment is encountered twice:

```json
{"msg":"collecting from producer","v":0,"name":"omicron-dev","level":20,"time":"2024-10-03T21:09:47.560547887Z","hostname":"devubuntu","pid":6084,"address":"[::1]:38053","producer_id":"a6458b7d-87c3-4483-be96-854d814c20de","component":"collection-task","collector_ip":"::1","collector_id":"39e6175b-4df2-4730-b11d-cbc1e60a2e78","component":"oximeter-agent","component":"oximeter"}
```

**Comment by davepacheco** [comment]
I think what's going on here is that we make heavy use the pattern of taking a logger passed to one component and customizing it for a subcomponent.  It's common to want to have the subcomponent use basically the same logger but with one extra key saying these messages are coming from the subcomponent.  I think at one point ages ago I incorrectly thought this property worked hierarchically, so that you'd get something like `"component": "a/b/c"`, and so we started using this.  But it doesn't work that way: you just get multiple key-value pairs with the same key.  The bright side is that it's possible to reconstruct the detail we want from the data that's there.  But we probably want a more composable pattern for doing this rather than just providing the same key multiple times.

**Comment by kika** [comment]
The dark side is that serde panics (if I remember correctly) when there're duplicate keys.  But, ofc, "there's a crate for that": https://docs.rs/serde-deserialize-duplicates/latest/serde_deserialize_duplicates/

Isn't array a more composable pattern or the order (like in "a/b/c") is important?

**Comment by davepacheco** [comment]
Serde doesn't panic in this case:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=707106e79a24c5edebf8cdc46d9e9bfe

Though it does return an error.  We read our logs almost exclusively with https://github.com/oxidecomputer/looker, which appears to parse the value as a `serde_json::Value` first before parsing _that_ into the struct:
https://github.com/oxidecomputer/looker/blob/76635da3e80a8818b71af335e7e8e4608a1fe2ca/src/main.rs#L402

Interestingly, that doesn't produce an error, though it of course loses all but one value for the field:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=43ddd003008b23707821accbeb1eed82

Anyway, yes, this is not ideal, but it hasn't been a priority to fix yet.

**Comment by jclulow** [comment]
> Interestingly, that doesn't produce an error, though it of course loses all but one value for the field:

Right, documents with duplicate keys in an object are not really JSON, as far as I recall.  The specification really only covers how to correctly handle a document in the event that there are no duplicates; everything else is basically undefined.
